### PR TITLE
Read "changelog" config key from "package.json" too

### PR DESCRIPTION
**Pull request by Turbo87**

### CODE REVIEW

**Comment by Turbo87** [comment]
@hzoo @gigabo any comments on this?

File: src/Configuration.js [hunk 1]
```diff
@@ -4,13 +4,13 @@ import path from "path";
...
-  const lernaPath = path.join(rootPath, "lerna.json");
-  const lernaJson = JSON.parse(fs.readFileSync(lernaPath));
+export function fromGitRoot(cwd) {
+  const rootPath = execSync("git rev-parse --show-toplevel", { cwd });
+  return fromPath(rootPath);
+}
 
-  const config = lernaJson.changelog;
+export function fromPath(rootPath) {
+  const config = fromPackageConfig(rootPath) || fromLernaConfig(rootPath);
```
Comment (gigabo) [inline src/Configuration.js]: I might let `lerna.json` have precedence over `package.json` if both are present.  It's more narrowly scoped.

Comment (Turbo87) [inline src/Configuration.js]: Hmm, my plan would have been to deprecate the lerna config option completely, but I'm fine with either solution as long as lerna itself is only optional :wink:

Comment (hzoo) [inline src/Configuration.js]: Hm I guess that's true, doesn't have to be in lerna.json but i'm ok with leaving it too

File: src/execSync.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import child from "child_process";
 
-export default function execSync(cmd) {
-  return child.execSync(cmd, {
-    encoding: "utf8"
-  }).trim();
+export default function execSync(cmd, options = {}) {
+  if (!options.encoding) {
+    options.encoding = "utf8";
```
Comment (emmenko) [inline src/execSync.js]: Any reason why to make this optional? When would you not need `utf8` encoding in our case?

File: test/Configuration.spec.js [hunk 3]
```diff
@@ -16,8 +21,55 @@ describe("Configuration", function() {
...
+      });
+
+      const result = fromPath(tmpDir);
+      expect(result.repo).toEqual("foo/bar");
+    });
+
+    it("reads the configuration from 'package.json'", function() {
+      fs.writeJsonSync(path.join(tmpDir, "package.json"), {
+        changelog: { repo: "foo/bar" },
+      });
```
Comment (emmenko) [inline test/Configuration.spec.js]: Nitpick: you can nest describes to make it more readable, then put the setup in `beforeEach`. The `it` blocks ideally should only contain the expectations.

```js
describe('fromPath', () => {
  describe('when lerna.json is present', () => {
    beforeEach(() => {
      fs.writeJsonSync(path.join(tmpDir, 'lerna.json'), {
        changelog: { repo: 'foo/bar' },
      })
    })
    describe('the changelog configuration is loaded', () => {
      let result
      beforeEach(() => {
        result = fromPath(tmpDir)
      })
      it('has correct repo value', () => {
        expect(result.repo).toEqual('foo/bar')
      })
    })
  })
  describe('when package.json is present', () => {
    // same as before
  })
})
```

File: src/execSync.js (hunk 2 above)
Comment (evocateur) [inline src/execSync.js]: If anything, this should be the default, with the `options` assigned over (not mutating the parameter value).

```js
const opts = Object.assign({
  encoding: "utf8",
}, options);

return child.execSync(md, opts).trim();
```

Comment (hzoo) [inline src/execSync.js]: sounds good

File: test/Configuration.spec.js (hunk 3 above)
Comment (Turbo87) [inline test/Configuration.spec.js]: I've just tried this, but due to the different scenarios that share no code the tests actually seem much harder to read after that refactoring 🤔

File: src/execSync.js (hunk 2 above)
Comment (Turbo87) [inline src/execSync.js]: adjusted the code to match the proposal of @evocateur

**Comment by hzoo** [comment]
@Turbo87 just need to add teh changelog 😄

**Comment by Turbo87** [comment]
seems a little meta, don't you think 😜

**Pull request merged**